mmcv/runner/hooks/logger/wandb.py

```diff
@@ -39,7 +58,29 @@ def before_run(self, runner):
             self.wandb.init(**self.init_kwargs)
         else:
             self.wandb.init()
-
+        
+        if self.config_path is not None:
+            if os.path.isdir(self.config_path):
+                copy_tree(self.config_path, self.wandb.run.dir)
+                for path_under_wandb, _, _ in os.walk(self.wandb.run.dir):
+                    self.wandb.save(
+                        glob_str=os.path.join(path_under_wandb,'*'),
+                        base_path=self.wandb.run.dir,
+                        policy='now'
+                    )
+            else:
+                if os.path.isfile(self.config_path):
+                    shutil.copy2(self.config_path, self.wandb.run.dir)
+                    self.wandb.save(
+                        glob_str=os.path.join(self.wandb.run.dir,'*'),
+                        base_path=self.wandb.run.dir,
+                        policy='now'
+                    )
+                else:
+                    raise FileNotFoundError(
+                        "No such file or directory: " + self.config_path
+                    )
```

Comment:
Should we move those lines to `before_run` because it will make `import_wandb` simpler and clearer.

Reply:
It is already in `before_run` function. Please check it. When we run all these lines, we need the `wandb.run.dir`. `wandb.run.dir` is defined at `wandb.init()`. So I have no other opinion to move.